### PR TITLE
Fixed #31870 -- Fixed crash when populating app registry with empty or without apps module.

### PR DESCRIPTION
**Pull request by iurisilvio**
ticket-31870

### CODE REVIEW

**Comment by felixxm** [comment]
@iurisilvio Thanks :+1: We should fix also deprecation warnings:

- _"RemovedInDjango41Warning: 'test_one' defines default_app_config = 'apps.explicit_default_config_without_apps.ExplicitDefaultConfigWithoutApps'. However, Django's automatic detection picked another configuration, **None**. You should move the default config class to the apps submodule of your application and, if this module defines several config classes, mark the default one with default = True. app_config = AppConfig.create(entry)"_,

- _"RemovedInDjango41Warning: 'test_two' defines default_app_config = 'apps.explicit_default_config_with_empty_apps.ExplicitDefaultConfigWithEmptyApps'. However, Django's automatic detection picked another configuration, **None**. You should move the default config class to the apps submodule of your application and, if this module defines several config classes, mark the default one with default = True. app_config = AppConfig.create(entry)"_

```

**Comment by iurisilvio** [comment]
I changed the message from

    However, Django's automatic detection picked another configuration, None.

to

    However, Django's automatic detection did not found a configuration.

Also, changed tests to validate the message instead of ignore.

File: tests/apps/tests.py [hunk 1]
```diff
@@ -520,3 +526,41 @@ def test_explicit_default_app_config_mismatch(self):
...
+            "ExplicitDefaultConfigWithEmptyApps'. However, Django's "
+            "automatic detection did not found a configuration. You should "
+            "move the default config class to the apps submodule of your "
+            "application and, if this module defines several config classes, "
+            "mark the default one with default = True."
+        )
+        with self.assertRaisesMessage(RemovedInDjango41Warning, msg):
+            with self.settings(INSTALLED_APPS=['apps.explicit_default_config_with_empty_apps']):
+                config = apps.get_app_config('explicit_default_config_with_empty_apps')
+            self.assertIsInstance(config, ExplicitDefaultConfigWithEmptyApps)
```
Comment (felixxm) [inline tests/apps/tests.py]: This assertion is unreachable.

File: tests/apps/tests.py [hunk 2]
```diff
@@ -520,3 +526,41 @@ def test_explicit_default_app_config_mismatch(self):
...
+            "ExplicitDefaultConfigWithoutApps'. However, Django's "
+            "automatic detection did not found a configuration. You should "
+            "move the default config class to the apps submodule of your "
+            "application and, if this module defines several config classes, "
+            "mark the default one with default = True."
+        )
+        with self.assertRaisesMessage(RemovedInDjango41Warning, msg):
+            with self.settings(INSTALLED_APPS=['apps.explicit_default_config_without_apps']):
+                config = apps.get_app_config('explicit_default_config_without_apps')
+            self.assertIsInstance(config, ExplicitDefaultConfigWithoutApps)
```
Comment (felixxm) [inline tests/apps/tests.py]: This assertion is unreachable.

File: django/apps/config.py [hunk 3]
```diff
@@ -91,7 +91,7 @@ def create(cls, entry):
         app_config_class = None
         app_name = None
         app_module = None
-
+        app_config_name = None
```
Comment (aaugustin) [inline django/apps/config.py:94]: Might want to move this line just after `app_config_class = None` and keep the blank line, which helps a bit the readability of this way-too-long function.

Comment (iurisilvio) [reply django/apps/config.py]: Done @aaugustin!

File: tests/apps/tests.py (hunk 2 above)
Comment (iurisilvio) [reply tests/apps/tests.py]: It was copied from the other tests, they have the same problem of unreachable assertion.

Comment (felixxm) [reply tests/apps/tests.py]: Yes, I've created PR #13289 to fix them.

**Pull request merged**